### ISSUE TITLE
Invalid model in QF_ABV

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#5736]
```
[566] % cvc4 --check-models -q small.smt2
Fatal failure within void CVC4::smt::CheckModels::checkModel(CVC4::smt::Model*, CVC4::context::CDList<CVC4::NodeTemplate<true> >*, bool) at src/smt/check_models.cpp:129
Internal error detectedSmtEngine::checkModel(): ERRORS SATISFYING ASSERTIONS WITH MODEL:
assertion:     k
simplifies to: false
expected `true'.
Run with `--check-models -v' for additional diagnostics.
Aborted
[567] % 
[567] % cat small.smt2
(declare-fun a () (Array (_ BitVec 32) (_ BitVec 32)))
(declare-fun b () (Array (_ BitVec 32) (_ BitVec 32)))
(declare-fun n () (_ BitVec 32))
(declare-fun o () (_ BitVec 32))
(declare-fun p () (Array (_ BitVec 32) (_ BitVec 32)))
(define-fun c () (_ BitVec 32) (_ bv0 32))
(define-fun d () (_ BitVec 32) (_ bv4 32))
(define-fun l () (Array (_ BitVec 32) (_ BitVec 32)) b)
(define-fun e () (_ BitVec 32) n)
(define-fun an () (Array (_ BitVec 32) (_ BitVec 32)) a)
(define-fun ao () (_ BitVec 32) (select an e))
(define-fun ap () (_ BitVec 32) o)
(define-fun aq () Bool (= ao ap))
(define-fun ar () Bool aq)
(define-fun f () (_ BitVec 32) (bvurem d ap))
(define-fun m () (Array (_ BitVec 32) (_ BitVec 32)) p)
(define-fun g () (Array (_ BitVec 32) (_ BitVec 32)) (store l f c))
(define-fun h () Bool (= m g))
(define-fun i () Bool (xor ar h))
(define-fun j () Bool (= ap c))
(define-fun k () Bool (xor i j))
(assert k)
(check-sat)
[568] % 
```
Commit: e0dfc0a

**Comment by muchang** [comment]
Another case:
```
[737] % cvc4 -q --check-models small.smt2
Fatal failure within void CVC4::smt::CheckModels::checkModel(CVC4::smt::Model*, CVC4::context::CDList<CVC4::NodeTemplate<true> >*, bool) at /local/suz-local/software/CVC4/src/smt/check_models.cpp:129
Internal error detectedSmtEngine::checkModel(): ERRORS SATISFYING ASSERTIONS WITH MODEL:
assertion:     (and (xor (distinct c (store b (bvsub #b00000000000000000000000000001000 a) #b00000000000000000000000000000001)) (distinct d #b00000000000000000000000000000000)) (= a e))
simplifies to: false
expected `true'.
Run with `--check-models -v' for additional diagnostics.
Aborted
[738] % 
[738] % cat small.smt2
(declare-fun a () (_ BitVec 32))
(declare-fun b () (Array (_ BitVec 32) (_ BitVec 32)))
(declare-fun c () (Array (_ BitVec 32) (_ BitVec 32)))
(declare-fun d () (_ BitVec 32))
(declare-fun e () (_ BitVec 32))
(assert (and (xor (distinct c (store b (bvsub (_ bv8 32) a) (_ bv1 32))) (distinct d (_ bv0 32))) (= a e)))
(check-sat)
[739] % 
```
Commit: e0dfc0a

**Comment by muchang** [comment]
Here is another related case: 
```
[599] % cvc4 -q --check-models --check-unsat-cores small.smt2
Fatal failure within void CVC4::smt::CheckModels::checkModel(CVC4::smt::Model*, CVC4::context::CDList<CVC4::NodeTemplate<true> >*, bool) at /local/suz-local/software/CVC4/src/smt/check_models.cpp:129
Internal error detectedSmtEngine::checkModel(): ERRORS SATISFYING ASSERTIONS WITH MODEL:
assertion:   (= g (store b (bvxor #b00000000000000000000000000000100 f) #b00000000000000000000000000000000))
simplifies to: false
expected `true'.
Run with `--check-models -v' for additional diagnostics.
Aborted
[600] % 
[600] % cat small.smt2
(declare-fun a () (Array (_ BitVec 32) (_ BitVec 32)))
(declare-fun b () (Array (_ BitVec 32) (_ BitVec 32)))
(declare-fun c () (_ BitVec 32))
(declare-fun d () (_ BitVec 32))
(declare-fun e () (_ BitVec 32))
(declare-fun f () (_ BitVec 32))
(declare-fun g () (Array (_ BitVec 32) (_ BitVec 32)))
(assert (= (= d e) (= (select a c) f)))
(assert (= g (store b (bvxor (_ bv4 32) f) (_ bv0 32))))
(check-sat)
[601] %
```

**Comment by mpreiner** [comment]
Fixed by #6660.
